tests/serac_odes.cpp

```diff
@@ -0,0 +1,645 @@
+// Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include <gtest/gtest.h>
+
+#include <array>
+#include <fstream>
+#include <functional>
+
+#include "mfem.hpp"
+
+#include "serac/numerics/expr_template_ops.hpp"
+#include "serac/physics/operators/odes.hpp"
+#include "serac/physics/operators/stdfunction_operator.hpp"
+
+using namespace serac;
+using namespace serac::mfem_ext;
+
+bool with_print_statements = true;
+
+DirichletEnforcementMethod enforcement_methods[3] = {DirichletEnforcementMethod::DirectControl,
+                                                     DirichletEnforcementMethod::RateControl,
+                                                     DirichletEnforcementMethod::FullControl};
+
+IterativeSolverOptions linear_options{.rel_tol     = 1.0e-12,
+                                      .abs_tol     = 1.0e-12,
+                                      .print_level = -1,
+                                      .max_iter    = 10,
+                                      .lin_solver  = LinearSolver::CG,
+                                      .prec        = {}};
+
+NonlinearSolverOptions nonlinear_options{.rel_tol = 1.0e-12, .abs_tol = 1.0e-12, .max_iter = 10, .print_level = -1};
+
+mfem::DenseMatrix M = []() {
+  mfem::DenseMatrix M(3, 3);
+  M(0, 0) = 2.0;
+  M(0, 1) = 1.0;
+  M(0, 2) = 0.0;
+
+  M(1, 0) = 1.0;
+  M(1, 1) = 4.0;
+  M(1, 2) = 1.0;
+
+  M(2, 0) = 0.0;
+  M(2, 1) = 1.0;
+  M(2, 2) = 2.0;
+  return M;
+}();
```

Comment:
I like this use of immediately-invoked lambdas.

Reply:
Agreed!

Reply:
it might be worth making a PR to mfem to add some convenience constructors of the form:

```cpp
template < int m, int n >
DenseMatrix(const double (&values)[m][n]) { ... }

template < int n >
Vector(const double (&values)[n]) { ... }
```

so we can just write `mfem::DenseMatrix A = {{1.0, 2.0, 3.0}, {4.0, 5.0, 6.0}}` and avoid having to use the lambda workaround